File: 344_Reverse_String/Reverse_String.go
func reverseString(s string) string {
    reverse_string := []rune(s)
    for head, tail := 0, len(s) - 1; head < tail ; head, tail = head + 1, tail - 1 {
        reverse_string[head], reverse_string[tail] = reverse_string[tail], reverse_string[head]
    }
    return string(reverse_string)
}

File: 344_Reverse_String/Reverse_String.py
class Solution(object):
    def reverseString(self, s):
        # return s[::-1]
        charList = list(s)
        head, tail = 0, len(charList) - 1
        while head < tail:
            charList[head], charList[tail] = charList[tail], charList[head]
            head, tail = head + 1, tail - 1
        return "".join(charList)
